src/NuGet.Core/NuGet.ProjectModel/Utf8JsonStreamReader.cs

```diff
@@ -104,6 +105,52 @@ internal void Skip()
             }
         }
 
+        internal IList<T> ReadObjectAsList<T>(IUtf8JsonStreamReaderConverter<T> streamReaderConverter)
+        {
+            if (TokenType != JsonTokenType.StartObject)
+            {
+                return Array.Empty<T>();
+
+            }
+            //We use JsonObjects for the arrays so we advance to the first property in the object which is the name/ver of the first library
+            Read();
+
+            if (TokenType == JsonTokenType.EndObject)
+            {
+                return Array.Empty<T>();
+            }
+
+            var listObjects = new List<T>();
+            do
+            {
+                listObjects.Add(streamReaderConverter.Read(ref this));
+                //At this point we're looking at the EndObject token for the object, need to advance.
+                Read();
+            }
+            while (TokenType != JsonTokenType.EndObject);
+            return listObjects;
+        }
+
+        internal void ReadArrayOfObjects<T1, T2>(IList<T2> objectList, IUtf8JsonStreamReaderConverter<T1> streamReaderConverter) where T1 : T2
+        {
+            if (objectList is null)
+            {
+                return;
```

Comment:
From an API design point of view, I don't understand why it would be acceptable to call this method with a null `objectList`, and for this method to no-op in that scenario. If there's data to be read.

Reply:
Yea, I changed it to just return the list.